File: src/Other/MicroPython/flyweight.py
class TextStyle:
    def __init__(self, font, size, color):
        self.font = font
        self.size = size
        self.color = color


class StyleFactory:
    def __init__(self):
        self.pool = {}

    def get(self, font, size, color):
        key = (font, size, color)
        if key not in self.pool:
            self.pool[key] = TextStyle(font, size, color)
        return self.pool[key]


factory = StyleFactory()
red1 = factory.get("Inter", 12, "red")
red2 = factory.get("Inter", 12, "red")
blue = factory.get("Inter", 12, "blue")
assert blue is not red1
print("styles={};shared={};text=ABC".format(
    len(factory.pool), "true" if red1 is red2 else "false"))

File: src/Web/Dart/flyweight.dart
class TextStyle {
  const TextStyle(this.font, this.size, this.color);

  final String font;
  final int size;
  final String color;
}

class StyleFactory {
  final Map<String, TextStyle> _pool = <String, TextStyle>{};

  TextStyle get(String font, int size, String color) {
    final key = '$font|$size|$color';
    return _pool.putIfAbsent(key, () => TextStyle(font, size, color));
  }

  int get count => _pool.length;
}

void main() {
  final factory = StyleFactory();
  final red1 = factory.get('Inter', 12, 'red');
  final red2 = factory.get('Inter', 12, 'red');
  final blue = factory.get('Inter', 12, 'blue');
  assert(blue.color == 'blue');
  print('styles=${factory.count};shared=${identical(red1, red2) ? 'true' : 'false'};text=ABC');
}
